src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Shapes = 'triangle' | 'rectangle' | 'circle';
+type Colors = 'red' | 'blue' | 'green';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'Side lengths must be positive numbers, greater than zero!',
+      );
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The provided sides do not form a valid triangle');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shapes = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Colors,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
There's a typo in the error message. It should be 'Radius must be a positive number, greater than zero'.